### PR TITLE
Bugfix: update sketch scene colors when theme changes

### LINKED ISSUE #2564: Sketch mode does not respect theme when switching
Steps to reproduce.

- go into sketch mode
- switch the theme
- witness the sadness

### PR DESCRIPTION
**Pull request by franknoirot**
Fixes #2564. We were calling something that attempted to update the client-side scene's colors, but upon investigation it wasn't doing so properly or thoroughly. This ensures it does so, and adds an E2E test to verify it.

### CODE REVIEW

**Comment by franknoirot** [comment]
Why is this diff absolutely busted?

**Comment by lf94** [comment]
Rebase on main / cherry-pick to main. I think you based on an old "[Migrate to XState v5 (](https://github.com/KittyCAD/modeling-app/pull/3847/commits/5f8d4f829444b9482d7e17a744ee23c06bf0413d)https://github.com/KittyCAD/modeling-app/pull/3735[)](https://github.com/KittyCAD/modeling-app/pull/3847/commits/5f8d4f829444b9482d7e17a744ee23c06bf0413d)" commit

File: src/clientSideScene/sceneEntities.ts [hunk 1]
```diff
@@ -1465,6 +1465,23 @@ export class SceneEntities {
...
+   * that represent each of the sketch segments, to get the
+   * latest value from `sceneInfra._theme`
+   */
+  updateSegmentBaseColor(newColor: Themes.Light | Themes.Dark) {
+    const newColorThreeJs = getThemeColorForThreeJs(newColor)
+    Object.values(this.activeSegments).forEach((group) => {
+      group.userData.baseColor = newColorThreeJs
+      group.traverse((child) => {
+        if (child instanceof Mesh) {
+          const mat = child.material as MeshBasicMaterial
```
Comment (lf94) [inline src/clientSideScene/sceneEntities.ts:1479]: Is there no way to guarantee this is a MeshBasicMaterial? To avoid `as` ?

Comment (franknoirot) [reply src/clientSideScene/sceneEntities.ts:1479]: I bet there is, good call

**Pull request closed without merging**